hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaArtifactManagement.java

```diff
@@ -98,20 +99,33 @@ private static Artifact checkForExistingArtifact(final String filename, final bo
     public Artifact create(final ArtifactUpload artifactUpload) {
         final String filename = artifactUpload.getFilename();
         final long moduleId = artifactUpload.getModuleId();
-        AbstractDbArtifact result = null;
 
         final SoftwareModule softwareModule = getModuleAndThrowExceptionIfThatFails(moduleId);
 
-        final Artifact existing = checkForExistingArtifact(filename, artifactUpload.overrideExisting(),
-                softwareModule);
+        final Artifact existing = checkForExistingArtifact(filename, artifactUpload.overrideExisting(), softwareModule);
 
         assertArtifactQuota(moduleId, 1);
         assertMaxArtifactSizeQuota(filename, moduleId, artifactUpload.getFilesize());
         assertMaxArtifactStorageQuota(filename, artifactUpload.getFilesize());
 
+        AbstractDbArtifact result = getOrCreateArtifact(artifactUpload);
```

Comment:
You could use the Optional in order to get rid of null checks:
```suggestion
        return getOrCreateArtifact(artifactUpload)
                .map(artifact -> storeArtifactMetadata(softwareModule, filename, artifact, existing)).orElse(null);
```

Reply:
Yes, looks much cleaner indeed, thank you for the hint! :+1: